src/features/posts/postSlice.ts

```diff
@@ -0,0 +1,60 @@
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Post } from '../../types/Post';
+import { getUserPosts } from '../../api/posts';
+
+export interface PostsUseState {
+  selectedPost: Post | null;
+  posts: Post[];
+  isLoading: boolean;
+  error: null | string;
+}
+
+const initialState: PostsUseState = {
+  selectedPost: null,
+  posts: [],
+  isLoading: false,
+  error: null,
+};
+
+export const fetchPosts = createAsyncThunk(
+  'posts/fetchPosts',
+  (userId: number) => {
+    return getUserPosts(userId);
+  },
+);
+
+export const postSlice = createSlice({
+  name: 'posts',
+  initialState: initialState,
+  reducers: {
+    setPosts: (state, action: PayloadAction<Post[]>) => {
+      state.posts = action.payload;
+    },
+    setSelectedPost: (state, action: PayloadAction<Post | null>) => {
+      state.selectedPost = action.payload;
+    },
+  },
+  extraReducers(builder) {
+    builder.addCase(fetchPosts.pending, state => {
+      state.posts = [];
+      state.isLoading = true;
+    });
+
+    builder.addCase(fetchPosts.rejected, state => {
+      state.error = 'Something went wrong!';
```

Comment:
Consider handling specific error messages in the `fetchPosts.rejected` case to provide more detailed feedback. This can help in debugging and improving user experience.